scripts/test_env.sh

```diff
@@ -90,24 +94,85 @@ function bulk_export {
 }
 
 function run_peer () {
-    PEER="$TEST/peers/$1"
+    PEER="$TEST/peers/iroha$1"
     mkdir -p "$PEER"
     STORAGE="$PEER/storage"
     mkdir -p "$STORAGE"
     KURA_BLOCK_STORE_PATH="$STORAGE"
-    TORII_P2P_ADDR="$HOST:${p2p_ports[$1]}"
-    TORII_API_URL="$HOST:${api_ports[$1]}"
-    TORII_TELEMETRY_URL="$HOST:${telemetry_ports[$1]}"
-    IROHA_PUBLIC_KEY=${public_keys[$1]}
-    IROHA_PRIVATE_KEY="{ \"digest_function\": \"ed25519\", \"payload\": \"${private_keys[$1]}\" }"
-    exec -a "$1" "$TEST/peers/iroha" "$2" > "$PEER/.log" & disown
+    TORII_P2P_ADDR="${hosts[$1]}:${p2p_ports[$1]}"
+    TORII_API_URL="${hosts[$1]}:${api_ports[$1]}"
+    TORII_TELEMETRY_URL="${hosts[$1]}:${telemetry_ports[$1]}"
+    IROHA_PUBLIC_KEY="${public_keys[$1]}"
+    IROHA_PRIVATE_KEY="${private_keys[$1]}"
+    exec -a "iroha$1" "$TEST/peers/iroha" "$2" > "$PEER/.log" & disown
+}
+
+function run_n_peers {
+    generate_genesis_key_pair
+    for peer in $(seq 0 $(($1-1)))
+    do
+       generate_p2p_port $peer
+       generate_api_port $peer
+       generate_telemetry_port $peer
+       generate_peer_key_pair $peer
+       hosts[$peer]="$HOST"
+    done
+    SUMERAGI_TRUSTED_PEERS="$(generate_trusted_peers $1)"
+    for peer in $(seq 1 $(($1-1)))
+    do
+        run_peer $peer
+    done
+    run_peer 0 --submit-genesis
```

Comment:
This isn't quite what I'd like to happen. Overwriting the same environment variable can have unforeseen consequences. Again one more reason to consider writing this logic in Python.